app/features/contacts/add-contact-drawer.tsx

```diff
@@ -0,0 +1,129 @@
+import { Check, Plus } from 'lucide-react';
+import { useState } from 'react';
+import { Button } from '~/components/ui/button';
+import {
+  Drawer,
+  DrawerContent,
+  DrawerHeader,
+  DrawerTitle,
+  DrawerTrigger,
+} from '~/components/ui/drawer';
+import { useToast } from '~/hooks/use-toast';
+import { SearchBar } from '../../components/search-bar';
+import { getErrorMessage } from '../shared/error';
+import type { PublicUser } from '../user/user';
+import { useUser } from '../user/user-hooks';
+import { ContactAvatar } from './contact-avatar';
+import { useContacts, useCreateContact, useSearchUsers } from './contact-hooks';
+
+export function AddContactDrawer() {
+  const [open, setOpen] = useState(false);
+  const { toast } = useToast();
+  const createContact = useCreateContact();
+  const { results, setSearchQuery } = useSearchUsers();
+  const existingContacts = useContacts();
+  const currentUser = useUser();
+
+  // Filter out contacts that are already in the user's contact list
+  // and also filter out the user's own profile
+  const filteredResults = results.filter((contact) => {
```

Comment:
could we do this filtering on db as well so we don't even pull ones that are not needed?